torch_xla/csrc/tensor_impl.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+#include <ATen/Type.h>
+#include <c10/core/TensorImpl.h>
+
+#include "tensor.h"
+
+namespace torch_xla {
+
+// Tensor implementation class used to be fed to the at::Tensor.
+// Its scope is just to handle an XLATensor.
+class TensorImpl : public c10::TensorImpl {
```

Comment:
Rename to `XLATensorImpl`, we shouldn't rely on namespaces to disambiguate.

Reply:
namespaces have been invented to stop people prefixing names, but OK, let's use C in this case ;)